<?xml version="1.0"?>
<launch>

  <param name="/use_sim_time"                              value="true"/>
  <!-- Camera parameters -->
  <arg name="sensor_id"               default="" />
  <arg name="sensor_name"             default="kinect2" />

  <!-- detection parameters -->
  <arg name="algorithm"               default="3dms" />
  <arg name="Backprojection_Mode"               default="HSVD" />
  <arg name="init_rois_txt_path"               default=" " />

  <!-- launch kinect2_bridge frames transform -->
  <include file="$(find kinect2_bridge)/launch/kinect2_bridge_rosbag.launch">
    <arg name="sensor_name"             value="$(arg sensor_name)" />
  </include>

		  
  <!-- Launch ground based people detection node -->
     <node pkg="detection" type="multiple_objects_detection_$(arg algorithm)_node" name="multiple_objects_detection_$(arg sensor_name)" output="screen" required="true">
   
     <rosparam command="load"                        file="$(find detection)/conf/multiple_objects_detection.yaml" />    

     <param name="Backprojection_Mode"                  value="$(arg Backprojection_Mode)" />
     <param name="init_rois_txt_path"                             value="$(arg init_rois_txt_path)" />

     <remap from="/kinect2_head/rgb_lowres/image" to="/$(arg sensor_name)/rgb_lowres/image" />
     <remap from="/kinect2_head/depth_lowres/image" to="/$(arg sensor_name)/depth_lowres/image" />
     <remap from="/kinect2_head/depth_lowres/camera_info" to="/$(arg sensor_name)/depth_lowres/camera_info" />
     <remap from="/kinect2_head/rgb_lowres/camera_info" to="/$(arg sensor_name)/rgb_lowres/camera_info" />
     <remap from="/kinect2_head/image2D_rois_from_gui" to="/$(arg sensor_name)/image2D_rois_from_gui" />
     <remap from="/kinect2_head/image2D_rois_from_file/image" to="/$(arg sensor_name)/image2D_rois_from_file/image" />


   </node>

</launch> 
